src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/ClaimCreationEventsStatusService.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.cmc.claimstore.events.claim;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CaseEvent;
+import uk.gov.hmcts.cmc.claimstore.repositories.CaseRepository;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimSubmissionOperationIndicators;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.function.BiFunction;
+
+@Component
+@ConditionalOnProperty(prefix = "feature_toggles", name = "async_eventOperations_enabled")
+public class ClaimCreationEventsStatusService {
+
+    private final CaseRepository caseRepository;
+
+    private static BiFunction<ClaimSubmissionOperationIndicators, CaseEvent, ClaimSubmissionOperationIndicators>
+        updateClaimSubmissionIndicatorWithEvent = (indicator, caseEvent) -> {
+            ClaimSubmissionOperationIndicators.ClaimSubmissionOperationIndicatorsBuilder updatedIndicator
+                = indicator.toBuilder();
+
+            switch (caseEvent) {
+                case PIN_GENERATION_OPERATIONS:
+                    updatedIndicator.defendantNotification(YesNoOption.YES).bulkPrint(YesNoOption.YES)
+                        .defendantPinLetterUpload(YesNoOption.YES);
```

Comment:
`staffNotification` is missing here.

Reply:
It will be more readable if you format something like (build one property on each line) :
```
updatedIndicator
                   .defendantNotification(YesNoOption.YES)
                   .bulkPrint(YesNoOption.YES)
                   .defendantPinLetterUpload(YesNoOption.YES)
                   .staffNotification(YesNoOption.YES);
```

Reply:
Done

Reply:
Good Catch 👍 . Which told me that I am missing unit test as well. @sabahirfan